### ISSUE TITLE
[FR] Ability to select metrics from which epoch (aka step) to display on UI metrics

### ISSUE THREAD

**Issue by imaluengo** [mlflow/mlflow#2436]
Usually I run many experiments i want to compare together. I run them almost with identical settings for a number (lets say 100) epochs. During training for every batch I do register a multitude of metrics and I name them with `T/...` to denote they are batch metrics. Then I do similar for epoch summarization metrics during validation and name them `V/...`:

```
mlflow.log_metrics({'T/accuracy':  0.9, 'T/loss': 0.2}, step=current_batch_number)
...
mlflow.log_metrics({'V/accuracy':  0.9, 'V/loss': 0.2}, step=current_epoch)
```
In the UI I can only see the metrics for the last batch/epoch of the run, which it no always (almost never) the one I want to use to compare to other experiments. So, to allow better comparison across experiments from the UI table without having to go into the curves plot, I'd like to suggest two quick modifications to MLFLow tracking UI summary table (e.g. http://<yourmlflowserver>/#/experiments/51):

1. Make a distinction in `mlflow.tracking` to allow submitting batch metrics and epoch metrics and distinguish them in the database without having to prepend them with some keyword. E.g.

```
mlflow.log_batch_metrics(....)
mlflow.log_epoch_metrics(...)
```

2. In the UI, for each individual run, and for the "epoch metrics", allow to select which epoch to display (and memorise it). As the number of runs grow, we could manually select the metrics from which epoch to display for each run in the tables so that we could compare future experiments against it very quickly. 

Basically, the ability to select the "best step" for each RUN so that those metrics are always visible and comparable across runs.

**Comment by smurching** [comment]
@imaluengo thanks for filing this! Seems like there are two feature requests that came up:

1. The ability to explicitly distinguish between metrics logged per batch, vs metrics logged per epoch. Is there a reason your workaround (prefixing the names of per-batch / per-epoch metrics with a string) doesn't suffice? I ask because it'd be a significant data model & REST API change to actually distinguish between a notion of "batch" vs "epoch" metrics (& add complexity to use cases where there's no notion of batches/epochs), so it may be hard to introduce that distinction as a core MLflow concept. However, I'd hope that we can address the underlying pain point using existing APIs (& if not, would love to understand how they fall short!)
2. The ability to compare values of metrics at the "best step" across runs. Here, would it make sense to log a separate metric on each run containing the value of that metric at the best step? e.g. "best_step_accuracy", "best_step_loss", etc? You could then compare values of that metric across runs, vs having to specify the best step across different runs.

**Comment by tanmoy7989** [comment]
Any updates on this? Was this feature added ?
